src/base/PortalContext.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, ReactNode, useState } from 'react';
+
+export interface PortalContextData {
+  portalElement: HTMLDivElement | null;
+  setPortalElement: ((element: HTMLDivElement) => void) | null;
+}
+
+export const portalContextDefaultValues = {
+  portalElement: null,
+  setPortalElement: null
+};
+
+export const PortalContext = createContext<PortalContextData>(
+  portalContextDefaultValues
+);
+
+export const PortalProvider: React.FC<ReactNode> = ({ children }) => {
```

Comment:
@EmiBemi Is this file okay or would it be preferred if everything in PortalContext.tsx was moved into portal.tsx?

Reply:
It's fine with me!